packages/swagger2openapi/index.js

```diff
@@ -307,7 +307,11 @@ function fixupRefs(obj, key, state) {
             }
             else if (inSchema && (options.refSiblings === 'allOf')) {
                 delete obj.$ref;
-                state.parent[state.pkey] = { allOf: [ { $ref: tmpRef }, obj ]};
+                if(typeof obj.description === 'string'){
```

Comment:
I was not sure how other properties should be treated, see commit message https://github.com/Mermade/oas-kit/pull/287/commits/3892da8bb4d4a0a499ca532375662a3cc12bd9ee. I am sure that description is a special case, thats why I implemented it as such. All other cases (like a nested properties object) still work the same way as before, and the way JSON schema defines.

Reply:
Why is `description` a special case, in OAS or JSON Schema?

Reply:
I think in terms of JSON Schema the final state using the allOf trick is valid. In terms of OAS, if we do not do this transformation all siblings of $ref are ignored, see https://swagger.io/docs/specification/using-ref/ at the end.
So maybe you are right, we do not have to consider all possible valid JSON Schema scenarios as input here. It's enough to only consider valid swagger inputs. Hence, there will never be a sibling of $ref called "properties" of type object. But we should move all siblings of $ref next to allOf (not only description sibling).
Was this more or less understandable :/ ? What do you think?

Reply:
There's two schools of thought on this, a one-element `allOf` with a `$ref` inside it to side-step the no-ref-siblings rule, or to use a multi-element `allOf` to indicate that both **this** (the `$ref`) and **that** (any other ex-sibling keywords) need to be evaluated.

You can probably tell from the existing code and the test outputs what my preference is. I'm not really interested in creating a special case just for `description`, or `required` or `additionalProperties` etc etc.

Reply:
The existing code does not produce useful output in my opinion. In swagger context, to avoid the "no-$ref-siblings" circumstances, the only way to not loose the siblings information (e.g. date, description etc) is to lift these siblings one level up and make them siblings of allOf. If we do the lift, e.g. date and description information for the property of question are not lost, and therefore e.g. rendered in html documentations. If we do not do the lift, these information (now part of allOf array) is simply ignored by swagger tools.
As far as I can tell other tools to this transformation as well, and I have read about that as well, e.g. here https://github.com/OAI/OpenAPI-Specification/issues/2033.

Reply:
But in your changed test output you are not "lifting" the sibling properties like `description` anywhere, you are simply burying a `$ref` in a one-element `allOf` and leaving `description` where it was.

If "swagger tools" ignore additional elements of an `allOf` that is their problem, and IMHO a bug.

One way out of this would be if you could find a good name for it, to specify a **new** value for `refSiblings` which uses the one-element `allOf` form (i.e. only the `$ref` gets moved into an `allOf` **if** we are inside a schemaObject).

Reply:
As far as I can tell, the existing refSiblings === 'allOf' logic is already only executed in schema context. We are using https://github.com/byu-oit/openapi-enforcer to validate the generated output. This validation fails, if I use refSiblings === 'allOf' without my fix. If I do not set refSiblings (leaving at the default value), then the generated output is valid, but the generated HTML preview (using redoc) does not contain any description information, see screenshot:
<img width="465" alt="Screenshot 2020-10-02 at 17 46 27" src="https://user-images.githubusercontent.com/1071386/94942941-465e5880-04d7-11eb-8ab6-0242450cb183.png">

The redoc behaviour seems ok to me, since the swagger docs state that any $ref siblings should be ignored.

This is why I thought it's legitimate to fix it that way. Because when I apply my fix, openapi enforcer validation is ok, and the rendered HTML doc using redoc renders the description.

Reply:
Now you're talking about other (non-swagger) tools. If **ReDoc** doesn't respect the second and subsequent branches of an `allOf` IMHO that's a bug in ReDoc.

If `openapi-enforcer` says an `allOf` with more than one element is not valid, then that is definitely a bug in `openapi-enforcer`.

Reply:
`npx openapi-enforcer-cli validate openapi.yaml`

``` 
One or more errors exist in the OpenApi definition
  at: components > schemas > a > allOf > 1
    Missing required property: type
```

That's definitely a bug in `openapi-enforcer`, `type` is not a required property in a schemaObject.

Reply:
I think openapi enforcer does its job. According to https://swagger.io/docs/specification/data-models/oneof-anyof-allof-not/#allof allOf holds an array of schemas. This is also what https://github.com/OAI/OpenAPI-Specification/blob/3.0.1/versions/3.0.1.md says, as far as I understand.

openapi-enforcer says that only (sub)schemas are valid as allOf elements (according to openapi spec). redoc correctly ignores any $ref siblings (according to swagger spec).